### ISSUE TITLE
Improve Rails' error handling

### ISSUE THREAD

**Issue by jgarber623-gov** [deptofdefense/move.mil#207]
From the Puma logs, we're seeing some things like:

```
Error during failsafe response: Missing template errors/not_found, application/not_found with {:locale=>[:en], :formats=>[:pdf], :variants=>[], :handlers=>[:raw, :erb, :html, :builder, :ruby]}. Searched in:
  * "/var/app/current/app/views"
```

I _think_ the `errors_controller` code isn't properly capturing 404s for non-HTML resources. This may be solvable by adding `via: :all` to the 404 route.

We should also improve the test code to attempt to load miscellaneous 404s.

From some quick searching, a couple URLs that might be helpful:

- https://stackoverflow.com/questions/42886234/error-when-trying-to-making-dynamic-error-page-in-rails-application#42886398
- https://stackoverflow.com/questions/21834377/solved-rails-4-custom-404-causes-postgresql-connection-failure-on-heroku
- https://stackoverflow.com/questions/13996259/testing-error-pages-in-rails-with-rspec-capybara
- https://agileleague.com/blog/rails-3-2-custom-error-pages-the-exceptions_app-and-testing-with-capybara/
- http://guides.rubyonrails.org/configuring.html

**Comment by jgarber623-gov** [comment]
After some digging around, it seems like using `exceptions_app` isn't the preferred way of handling application errors. rails/rails#19913 and rails/rails#17815 have some details as to why.

**Comment by jgarber623-gov** [comment]
Resolved by #268.
